feat!: Improve Pull Request Template & Contributing Guidelines (#750)
argoproj/argo-helm: CONTRIBUTING.md

Argo Helm is a collection of **community maintained** charts. Therefore we rely on you to test your changes sufficiently.

# Pull Requests

All submissions, including submissions by project members, require review. We use GitHub pull requests for this purpose. Consult [GitHub Help](https://help.github.com/articles/about-pull-requests/) for more information on using pull requests. See the above stated requirements for PR on this project.

## Versioning

Each chart's version follows the [semver standard](https://semver.org/). New charts should start at version `1.0.0`, if it's considered stable. If it's not considered stable, it must be released as [prerelease](#prerelease).

Any breaking changes to a chart (backwards incompatible) require:

  * Bump of the current Major version of the chart
  * State possible manual changes for this chart version in the `Upgrading` section of the chart's `README.md.gotmpl` ([See Upgrade](#upgrades))

### Immutability

Each release for each chart must be immutable. Any change to a chart (even just documentation) requires a version bump. Trying to release the same version twice will result in an error.

> ### Artifacthub Annotations

Reviewer: ```suggestion
### Artifact Hub Annotations
```

> Since we release our charts on Artifacthub we encourage making use of the provided chart annotations for Artifacthub.

Reviewer: ```suggestion
Since we release our charts on Artifact Hub we encourage making use of the provided chart annotations for Artifact Hub.
```

  * [https://artifacthub.io/docs/topics/annotations/helm/](https://artifacthub.io/docs/topics/annotations/helm/)

#### Changelog

> We want to deliver transparent chart releases for our chart consumers. Therefor we require a changelog per new chart release.

Reviewer: Use existing variant for this word (see line 3)
```suggestion
We want to deliver transparent chart releases for our chart consumers. Therefore we require a changelog per new chart release.
```
